### ISSUE TITLE
Next.js and Payload API route error: Response body object should not be disturbed or locked

### ISSUE THREAD

**Issue by 58bits** [payloadcms/payload#4642]
### Link to reproduction

https://github.com/infonomic/payload-custom-server

### Describe the Bug

In both the Payload custom server example, and the current website template and when using Next.js App router, if there is an 'api' route handler in Next.js - for example...

```
.
├── layout.tsx
├── api
│   └── set-theme
│       └── route.ts
└── page.tsx
```
.. and the route handler is a POST, PUT, PATCH request with a `body` payload, then Next.js will throw the following error:

```
TypeError: Response body object should not be disturbed or locked
    at extractBody (node:internal/deps/undici/undici:4323:17)
    at new _Request (node:internal/deps/undici/undici:5272:48)
    at new NextRequest (/payload-custom-server/node_modules/.pnpm/next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/src/server/web/spec-extension/request.ts:24:10)
    at Function.fromNodeNextRequest (/payload-custom-server/node_modules/.pnpm/next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/src/server/web/spec-extension/adapters/next-request.ts:98:12)
```
The error is being caused because of the shared `/api` route prefix between Payload and Next.js. 

With Payload routes / init registered first, Payload middleware is being applied before the route is rejected, and passed on to Next.js - which then errors as above because the body stream has already been read by Payload (which is what [undici](https://github.com/nodejs/undici/blob/35b049b7f9b1f0e597e726a82e2f6386d0ba2947/lib/fetch/body.js#L191) is complaining about).

The current 'fix' or workaround, is to either rename the `/api` path prefix in Payload via `payload.config.ts`  and the `routes` option, for example...
```
routes: {
  admin: '/admin',
  api: '/papi',
},
```
Or to rename the `api` folder (which is there by convention) inside the Next.js App router app folder - for example from `api` to `napi`

```
.
├── layout.tsx
├── napi
│   └── set-theme
│       └── route.ts
└── page.tsx
```

### To Reproduce

The repo above includes two buttons with onClick handlers on a `/test-page` route that will call test POST and test GET route handlers in Next.js. 

Remove the 'fix' from `payload.config.ts` - i.e. rename the Payload api path prefix back to `/api` and then run the project `pnpm install` `pnpm dev` 

When you visit `/test-page` - and click on the 'Test POST API Call'  button you'll see the error.

### Payload Version

2

**Comment by paulpopus** [comment]
@58bits I added it to my todo to look into updating the readme in that repo

**Comment by jacobsfletch** [comment]
@paulpopus why was this issue closed? I agree with @58bits here:

> This might not possible - but the 'nice' thing to do would be for Payload / Express to reject the route as a 'miss' instead of processing the request via middleware before letting 'other' handlers take over - like Next.js.

Is this possible? This would be the best solution imo. For the record, our [move to Next.js](https://github.com/payloadcms/payload/discussions/4202) will completely solve this, but it might be wise for us to do one of the following things in the meantime:

1. Look into this above solution
2. Rename our Next.js API route handlers as suggested and add documentation to the README (I don't love this solution, because it means there's a "gotcha" to look out for)
3. Do nothing and wait for 3.0, using this issue for others to find so they can wire in their own workaround

At the very least we need to update the README, but since that has not been done yet I am going to re-open this issue.

**Comment by paulpopus** [comment]
Hey @58bits sorry I didn't mean to close this issue on you entirely without a resolution.

We discussed this internally and we agreed on making changes to documentation (inside Readme and each Nextjs API route) as well as default configuration of the templates https://github.com/payloadcms/payload/pull/4725

This issue will be more elegantly fixed in 3.0
